pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -94,6 +98,8 @@ func NewProbe(cfg *ddebpf.Config) (*Probe, error) {
 	probe.mapBuffers.iterationRestartDetectionEntries = ddconfig.SystemProbe.GetInt("ebpf_check.entry_count.entries_for_iteration_restart_detection")
 	probe.entryCountMaxRestarts = ddconfig.SystemProbe.GetInt("ebpf_check.entry_count.max_restarts")
 
+	probe.mphCache = newMapProgHelperCache()
```

Comment:
Could we avoid creating this here if the `for_each_elem` helper is not available, and then handle `mphCache` being `nil` elsewhere?

Reply:
good idea, done in the latest push